addons/web/static/tests/core/position_hook_tests.js

```diff
@@ -150,6 +150,42 @@ QUnit.test("has no effect when component is destroyed", async (assert) => {
     );
 });
 
+QUnit.test("is positioned relative to its containing block", async (assert) => {
+    assert.expect(4);
+    // offset the container
+    container.style.margin = "15px";
+    let pos1, pos2;
+    let TestComp = getTestComponent({
+        onPositioned: (el, pos) => {
+            pos1 = pos;
+        },
+    });
+    let popper = await mount(TestComp, container);
+
+    const popBox1 = document.getElementById("popper").getBoundingClientRect();
+    destroy(popper);
+    document.getElementById("reference").remove();
+
+    // make container the containing block instead of the viewport
+    container.style.contain = "layout";
+
+    TestComp = getTestComponent({
+        onPositioned: (el, pos) => {
+            pos2 = pos;
+        },
+    });
+    popper = await mount(TestComp, container);
+    const popBox2 = document.getElementById("popper").getBoundingClientRect();
+    destroy(popper);
+
+    // best positions are not the same relative to their containing block
+    assert.notEqual(pos1.top, pos2.top);
+    assert.notEqual(pos1.left, pos2.left);
+    // best positions are the same relative to the viewport
+    assert.equal(popBox1.top, popBox2.top);
+    assert.equal(popBox1.left, popBox2.left);
+});
```

Comment:
What do you think about avoiding `assert.notEqual` (which I don't like as it is not precise) with something similar to the following (I did not try) ?
```suggestion
    const margin = 15;
    container.style.margin = `${margin}px`;
    const positions = [];
    const TestComp = getTestComponent({
        onPositioned: (el, pos) => {
            positions.push(pos);
        },
    });
    let popper = await mount(TestComp, container);
    const popBox1 = document.getElementById("popper").getBoundingClientRect();
    destroy(popper);
    document.getElementById("reference").remove();

    // make container the containing block instead of the viewport
    container.style.contain = "layout";
    popper = await mount(TestComp, container);
    const popBox2 = document.getElementById("popper").getBoundingClientRect();
    destroy(popper);

    // best positions are not the same relative to their containing block
    const [pos1, pos2] = positions;
    assert.equal(pos2.top, pos1.top + margin);
    assert.equal(pos2.left, pos1.left + margin);
    // best positions are the same relative to the viewport
    assert.equal(popBox1.top, popBox2.top);
    assert.equal(popBox1.left, popBox2.left);
});
```